### PR TITLE
Trade Component Refresh

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [X] Bug fix

### Linked issue

https://cybrid.atlassian.net/browse/CYB-1177

### Why

- Integrate Auto Init module inside the component (Cybrid.fiat)
- Migrate design to new one with Figma updates
- Migrate old BigDecimal to use CDecimal class with support for easy use
- Fix UI with correct UI States (like all components)
- Fix problem with modals to be two at the same time
- Fix bug with modals wit ui states to don’t override state in Polling
- Fix quantities according correct AssetBankModel
- Add test for ViewModel (Legacy code has this file out of the scope)
- Apply visual changes to DemoApp to handle the component 

### Evidence

https://user-images.githubusercontent.com/7268597/236880206-09d54b5f-b8a9-4154-9646-8105f802c8c0.mov

### CODE REVIEW

File: CybridSDK/Components/Trade/View/TradeViewModel.swift [hunk 1]
```diff
@@ -102,68 +103,179 @@ class TradeViewModel: NSObject, ListPricesItemDelegate {
...
-}
 
-extension TradeViewModel: UIPickerViewDelegate, UIPickerViewDataSource {
-
-    public func numberOfComponents(in pickerView: UIPickerView) -> Int {
-        return 1
+    internal func calculatePreQuote() {
+
+        let assetCode = currentAsset.value?.code ?? ""
+        let pairAssetCode = currentPairAsset.value?.code ?? ""
```
Comment (brent-cybrid) [inline CybridSDK/Components/Trade/View/TradeViewModel.swift]: These are referred to as `asset` and `counter_asset`.

File: CybridSDK/Utilities/Formatters/AssetFormatter.swift [hunk 2]
```diff
@@ -35,12 +36,12 @@ struct AssetFormatter {
 
         // Sanity check in intValue is zero, has to be removed
         if intValue == "0" { intValue = "" }
-        if asset.type == .crypto {
+        /*if asset.type == .crypto {
             decimalValue = decimalValue.removeLeadingZeros()
-        }
+        }*/
```
Comment (brent-cybrid) [inline CybridSDK/Utilities/Formatters/AssetFormatter.swift]: Remove

File: CybridSDKTests/Utilities/Formatters/AssetFormatterTest.swift [hunk 3]
```diff
@@ -427,12 +443,111 @@ class AssetFormatterTest: XCTestCase {
...
+            price: price.buyPrice!,
+            base: .fiat)
+        let tradeValueCDecimal = CDecimal(tradeValue)
+        let tradeBase = AssetFormatter.forBase(AssetBankModel.bitcoin, amount: tradeValueCDecimal)
+        let tradeFormatted = AssetFormatter.format(AssetBankModel.bitcoin, amount: tradeBase)
+
+        // -- Then
+        XCTAssertEqual(tradeValue, "495076")
+        XCTAssertEqual(tradeBase, "0.00495076")
+        XCTAssertEqual(tradeFormatted, "0.00495076 ₿")
```
Comment (brent-cybrid) [inline CybridSDKTests/Utilities/Formatters/AssetFormatterTest.swift]: The B should be in the front, not the end.

File: CybridSDK/Components/Trade/View/TradeViewModel.swift (hunk 1 above)
Comment (nalancer08) [reply CybridSDK/Components/Trade/View/TradeViewModel.swift]: Done!

File: CybridSDK/Utilities/Formatters/AssetFormatter.swift (hunk 2 above)
Comment (nalancer08) [reply CybridSDK/Utilities/Formatters/AssetFormatter.swift]: Done

File: CybridSDKTests/Utilities/Formatters/AssetFormatterTest.swift (hunk 3 above)
Comment (nalancer08) [reply CybridSDKTests/Utilities/Formatters/AssetFormatterTest.swift]: Done

File: CybridSDK/Utilities/Formatters/AssetFormatter.swift [hunk 4]
```diff
@@ -93,15 +91,51 @@ struct AssetFormatter {
     /// - 10.00 --> $10.00 USD
     /// - 10.50 --> $10.50 USD
     /// - 1000 --> $1,000 USD
-    /// - 90071992.54740991 --> ₿90,071,992.54740991
+    /// - 90071992.54740991 --> 90,071,992.54740991 ₿
```
Comment (brent-cybrid) [inline CybridSDK/Utilities/Formatters/AssetFormatter.swift]: This comment is wrong I believe

**Review by brent-cybrid — APPROVED** [review]

File: CybridSDK/Utilities/Formatters/AssetFormatter.swift (hunk 4 above)
Comment (nalancer08) [reply CybridSDK/Utilities/Formatters/AssetFormatter.swift]: Done!

**Pull request merged**